src/wikibase.py

```diff
@@ -267,58 +272,123 @@ def get_concept(
             concept.alternative_labels = list(all_positive_labels)
             concept.negative_labels = list(all_negative_labels)
 
+        if include_ancestry:
+            concept.ancestry = self.get_ancestry(wikibase_id)
+
         return concept
 
-    def get_concept_ids(self) -> list[WikibaseID]:
+    def get_ancestry(self, wikibase_id: WikibaseID) -> list[WikibaseID]:
+        """Fetch the complete ancestry of a concept"""
+        # Resolve any redirects first
+        wikibase_id = self._resolve_redirect(wikibase_id)
+
+        response = self.session.get(
+            url=self.api_url,
+            params={
+                "action": "wbgetentities",
+                "format": "json",
+                "ids": wikibase_id,
+                "props": "claims",
+            },
+        ).json()
+
+        entity = response["entities"][wikibase_id]
+        ancestry = [wikibase_id]
+        if "claims" in entity:
+            for claim in entity["claims"].values():
+                for statement in claim:
+                    if statement["mainsnak"]["snaktype"] == "value":
+                        property_id = statement["mainsnak"]["property"]
+                        value = statement["mainsnak"]["datavalue"]["value"]
+                        if property_id == self.subconcept_of_property_id:
+                            resolved_id = self._resolve_redirect(value["id"])
+                            ancestry.append(resolved_id)
+                            ancestry.extend(self.get_ancestry(resolved_id))
```

Comment:
Is there potential that this could infinite loop if there is a cylical 'subconcept of' relationship?

Reply:
i think the escape hatch can probably be built using a `max_number_of_levels` parameter with a sensible default? e.g. if it's 10, even if there's a loop it'll only try 10 times before stopping anyway

Reply:
great idea! i've made the change

the `wikibase._get_recursive_relationships()` method now takes a `max_depth` kwarg (default 50). at that depth, we should raise a warning and break the loop, returning the values accumulated so far